packages/engine/lib/error/src/future.rs

```diff
@@ -268,7 +179,7 @@ pub trait FutureExt: Future + Sized {
     #[track_caller]
     fn attach_message<M>(self, message: M) -> FutureWithMessage<Self, M>
     where
-        M: fmt::Display + fmt::Debug + Send + Sync + 'static;
```

Comment:
How come you keep swapping between these?

Reply:
This is because of the macro itself:
```
$bound:ident $(+ $bounds:ident)
```

If this would be `$bound:ty` or `$bound:path` the `+` would not be allowed.

Reply:
Alright